### PR TITLE
AB#187969: Polymer v1.0

### PR DESCRIPTION
**Pull request by maddyparker01**
# Overview

polymers

## Azure Boards

- [AB#187969](https://dev.azure.com/UniversityOfNottingham/bd2de5ba-2a41-4bdc-a444-61e6b706132a/_workitems/edit/187969)

### CODE REVIEW

File: Webapp/sources/templates/sketcher_reload.html [hunk 1]
```diff
@@ -39,6 +39,13 @@ <h4>{{ reaction.reaction_id }}</h4>
         <p>Please sketch or upload your reaction to begin. Click <a target="_blank" href="{{ url_for('main.marvin_js_help') }}">here</a> to view our Marvin JS help guide.</p>
         <p>Please note compounds drawn above or underneath the arrow will not be recognised.</p>
     </div>
+    <div>
+        <form action="{{ url_for('reaction_table.process') }}" method="post">
+            <label> Polymer Mode?
+                <input type="checkbox" id="polymer-mode-select" name="polymer-mode-select" onclick="savePolymerMode()" {% if reaction.polymer_mode %} checked {% endif %}>
```
Comment (JoeHeeley2) [inline Webapp/sources/templates/sketcher_reload.html:45]: does this need changing to reflect the new db migration?

```suggestion
                <input type="checkbox" id="polymer-mode-select" name="polymer-mode-select" onclick="savePolymerMode()" {% if reaction.reaction_type %} checked {% endif %}>
```

File: Webapp/migrations/versions/7705cc9f79e5_polymer.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+def downgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table("Reaction", schema=None) as batch_op:
+        batch_op.drop_column("reaction_rxn")
+        batch_op.drop_column("polymerisation_type")
+        batch_op.drop_column("reaction_type")
+        batch_op.drop_column("reaction_image")
+
+    # Drop the ENUM type
+    reaction_type_enum = sa.Enum("STANDARD", "POLYMER", name="reactiontype")
```
Comment (JoeDavies-6) [inline Webapp/migrations/versions/7705cc9f79e5_polymer.py:76]: Have you tested db upgrade/downgrade works? Looks like it should do from this.

File: Webapp/sources/blueprints/reaction_table/routes.py [hunk 3]
```diff
@@ -160,13 +191,20 @@ def process():
     else:
         sol_rows = services.solvent.get_workbook_list(workbook)
 
-    r_class, r_classes = services.reaction_classification.classify_reaction(
-        reactants_smiles_list, products_smiles_list
-    )
+    # change rxn table in polymer mode
+    if polymer_mode == "true":
```
Comment (JoeHeeley2) [inline Webapp/sources/blueprints/reaction_table/routes.py:195]: does this need to be changed to lowercase with polymer_mode.lower()?

**Comment by JoeDavies-6** [comment]
So long as all references to reaction.polymer_mode have been replaced with reaction.reaction_type, LGTM.

**Review by JoeDavies-6 — APPROVED** [review]

**Review by JoeDavies-6 — APPROVED** [review]

**Review by JoeHeeley2 — APPROVED** [review]

**Pull request merged**